### DISCUSSION TITLE
Tailwind css pseudo-elements don't work correctly.

### DISCUSSION THREAD

**Discussion by ElathonWeb** [tailwindlabs/tailwindcss#10381] [Help] [answered]
I'm creating a small project in React-native and for styling I'm using Tailwind css. I just started using react-native and Tailwind css so I don't know much. I was following the documentation, when I came across trying to use the :after pseudo-element, but the result was not what I hoped. The style was applied as if the pseudo element weren't there. So I tried with another pseudo-element: first-letter: but the result was the same.

**This is my code:**
```node
<View className="block">
    <Text className="after:content-['*'] after:ml-0.5 after:text-red-500 inline-block text-base ml-1">Name</Text>
    <TextInput
        className="bg-slate-50 p-4 rounded-lg mt-1 text-xl text-gray-500"
    />
</View>
```

**This is my result:**
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://i.imgur.com/uGbQxqH.png">
  <source media="(prefers-color-scheme: light)" srcset="https://i.imgur.com/uGbQxqH.png">
  <img alt="My result" src="https://i.imgur.com/uGbQxqH.png">
</picture>

**This is what I expected instead:**
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://i.imgur.com/5ZR2Fih.png">
  <source media="(prefers-color-scheme: light)" srcset="https://i.imgur.com/5ZR2Fih.png">
  <img alt="Expected result" src="https://i.imgur.com/5ZR2Fih.png">
</picture>

**package.json**
```json
"scripts": {
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "web": "expo start --web"
},
"dependencies": {
    "@react-navigation/native-stack": "^6.9.8",
    "expo": "~47.0.12",
    "expo-status-bar": "~1.4.2",
    "nativewind": "^2.0.11",
    "react": "18.1.0",
    "react-native": "0.70.5",
    "react-native-heroicons": "^3.2.0",
    "react-native-safe-area-context": "^4.5.0",
    "react-native-svg": "13.4.0"
},
"devDependencies": {
    "@babel/core": "^7.12.9",
    "tailwindcss": "^3.2.4",
    "tailwindcss-pseudo-elements": "^2.0.0"
},
"private": true
```

**Comment by adamwathan** [comment, marked as answer]
Hey! I've never used React Native but a quick search suggests that React Native just [doesn't support pseudo-elements at all](https://stackoverflow.com/questions/61430672/is-there-any-way-to-use-pseudo-elements-on-react-native-with-styled-components). Also Tailwind CSS itself doesn't work with React Native either — it looks like you are using [NativeWind](https://www.nativewind.dev/) which is not the same project. Best to direct questions about that project to the people in that community 👍

**Reply by ElathonWeb** [reply]
Now I understand many more things. I read a little more and indeed it is as you said. At first I didn't understand the difference between Tailwind CSS and NativeWind. Thank you very much. 😅
